podman/domain/manifests.py

```diff
@@ -0,0 +1,247 @@
+            raise NotFound(body["cause"], response=response, explanation=body["message"])
+        raise APIError(body["cause"], response=response, explanation=body["message"])
+
+    def remove(self, digest: str) -> None:
+        """Remove Image digest from manifest list.
+
+        Args:
+            digest: Image digest to be removed. Should a full Image reference be provided
+                the digest will be parsed out.
+
+        Raises:
+            ImageNotFound: When the Image could not be found
+            APIError: When service reports an error
+        """
+        if "@" in digest:
+            _, digest = digest.split("@")
+
+        response = self.client.delete(f"/manifests/{self.quoted_name}", params={"digest": digest})
+
+        if response.status_code == requests.codes.okay:
+            return self.reload()
+
+        body = response.json()
+        if response.status_code == requests.codes.not_found:
+            raise ImageNotFound(body["cause"], response=response, explanation=body["message"])
+        raise APIError(body["cause"], response=response, explanation=body["message"])
+
+    def reload(self) -> None:
+        """Refresh this object's data from the service."""
+        latest = self.manager.get(self.name)
+        self.attrs = latest.attrs
+
+
+class ManifestsManager(Manager):
+    """Specialized Manager for Manifest resources."""
+
+    resource = Manifest
+
+    def __init__(self, client: APIClient):
+        """Initiate ManifestsManager object.
+
+        Args:
+            client: Connection to Podman service.
+        """
+        super().__init__(client)
+
+    def exists(self, key: str) -> bool:
+        key = urllib.parse.quote_plus(key)
+        response = self.client.get(f"/manifests/{key}/exists")
+        return response.status_code == requests.codes.no_content
+
+    def create(
+        self,
+        names: List[str],
+        images: Optional[List[Union[Image, str]]] = None,
+        all: Optional[bool] = None,  # pylint: disable=redefined-builtin
+    ) -> Manifest:
+        """Create a Manifest.
+
+        Args:
+            names: Identifiers to be added to the manifest. There must be at least one.
+            images: Images or Image identifiers to be included in the manifest.
+            all: When True, add all contents from images given.
+
+        Raises:
+            ValueError: When no names are provided.
+            NotFoundImage: When a given image does not exist.
+        """
+        if names is None or len(names) == 0:
+            raise ValueError("At least one manifest name is required.")
+
+        params = {"name": names}
+        if images is not None:
+            params["image"] = list()
+            for item in images:
+                if isinstance(item, Image):
+                    item = item.attrs["RepoTags"][0]
+                params["image"].append(item)
+
+        if all is not None:
+            params["all"] = all
+
```

Comment:
Should you check if ***both*** `all` and `images` are `None`?

Reply:
In this case, `all` and `names` are not mutually exclusive.